wpilibj/src/main/java/edu/wpi/first/wpilibj/simulation/FlywheelSim.java

```diff
@@ -0,0 +1,80 @@
+import edu.wpi.first.wpilibj.util.Units;
+import edu.wpi.first.wpiutil.math.Matrix;
+import edu.wpi.first.wpiutil.math.numbers.N1;
+
+public class FlywheelSim extends LinearSystemSim<N1, N1, N1> {
+
+  private final DCMotor m_motor;
+  private final double m_gearing;
+
+  /**
+   * Create a LinearSystemSimulator. This simulator uses an {@link LinearSystem} to simulate
+   * the state of the system. In simulationPeriodic, users should first set inputs from motors, update
+   * the simulation and write simulated outputs to sensors.
+   *
+   * @param flywheelPlant      The flywheel system being controlled. This system can be created
+   *                           using {@link LinearSystemId#createFlywheelSystem(DCMotor, double, double)}
+   *                           or {@link LinearSystemId#identifyVelocitySystem(double, double)}
+   * @param flywheelGearbox    The DCMotor representing the motor driving the flywheel.
+   * @param gearing            The reduction between motor and drum, as output over input.
+   *                           In most cases this is greater than one.
+   * @param measurementStdDevs Standard deviations of measurements. Can be null if addNoise is false.
+   */
+  public FlywheelSim(LinearSystem<N1, N1, N1> flywheelPlant, DCMotor flywheelGearbox,
+                     double gearing, Matrix<N1, N1> measurementStdDevs) {
+    super(flywheelPlant, measurementStdDevs);
+    this.m_motor = flywheelGearbox;
+    this.m_gearing = gearing;
+  }
+
+  /**
+   * Create a LinearSystemSimulator. This simulator uses an {@link LinearSystem} to simulate
+   * the state of the system. In simulationPeriodic, users should first set inputs from motors, update
+   * the simulation and write simulated outputs to sensors.
+   *
+   * @param flywheelGearbox    The DCMotor representing the motor driving the flywheel.
+   * @param gearing            The reduction between motor and drum, as output over input.
+   *                           In most cases this is greater than one.
+   * @param moiKgMetersSquared The moment of inertia J of the flywheel. This can be
+   *                           calculated with CAD. If J is unknown, {@link #FlywheelSim(LinearSystem, DCMotor, double, Matrix)}
+   *                           using FRC-Characterization's kV and kA may be better.
+   * @param measurementStdDevs Standard deviations of measurements. Can be null if addNoise is false.
+   */
+  public FlywheelSim(DCMotor flywheelGearbox, double gearing,
+                     double moiKgMetersSquared, Matrix<N1, N1> measurementStdDevs) {
+    super(LinearSystemId.createFlywheelSystem(flywheelGearbox, moiKgMetersSquared, gearing),
+        measurementStdDevs);
+    this.m_motor = flywheelGearbox;
+    this.m_gearing = gearing;
+  }
+
+  public double getFlywheelVelocityRadPerSec() {
+    return getOutput(0);
+  }
+
+  public double getFlywheelVelocityRPM() {
+    return Units.radiansPerSecondToRotationsPerMinute(getOutput(0));
+  }
```

Comment:
If this were to follow the C++ code, it should be
```suggestion
  public double getVelocityRadPerSec() {
    return getOutput(0);
  }

  public double getVelocityRPM() {
    return Units.radiansPerSecondToRotationsPerMinute(getOutput(0));
  }
```